tv_apps/remote-control/js/app.js

```diff
@@ -0,0 +1,144 @@
+      this.keyNav.on('esc', this._handleBack.bind(this));
+
+      this.config = new Config();
+      this.config.start();
+      this.config.once('ready', this._statusCheck.bind(this));
+      this.config.on('changed', this._onSettingsChanged.bind(this));
+    },
+
+    stop: function() {
+      this.config.stop();
+      this.config = null;
+
+      this.keyNav.uninit();
+      this.keyNav = null;
+
+      this.activeSection = '';
+      this.sections = [];
+    },
+
+    _statusCheck: function() {
+      var setVisible = function(elemId, visible) {
+        var elem = document.getElementById(elemId);
+        if (visible) {
+          elem.classList.add('visible');
+        } else {
+          elem.classList.remove('visible');
+        }
+      };
+
+      var enabled = this.config.settings.enabled;
+      var hasConnection = !!this.ip;
+      setVisible('location', enabled && hasConnection);
+      setVisible('off-line-message', enabled && !hasConnection);
+      setVisible('disabled-message', !enabled);
+
+      var mainSection = this.sections[MAIN_SECTION];
+      if (enabled && hasConnection) {
+        mainSection.focus(document.getElementById(DEFAULT_ELEMENT));
+      } else {
+        mainSection.unfocus();
+        mainSection.focus();
+      }
+    },
+
+    _onSettingsChanged: function(name, value) {
+      switch(name) {
+        case 'server-ip':
+          this._updateIP(value);
+          break;
+      }
+      this._statusCheck();
+    },
+
+    _switchSection: function(sectionId) {
+      if (!sectionId) {
+        sectionId = MAIN_SECTION;
+      }
+      this.activeSection.hide();
+      this.activeSection = this.sections[sectionId].show();
+    },
+
+    _updateIP: function(ip) {
+      this.ip = ip;
+      document.getElementById('ip').textContent = this.ip;
+    },
+
+    _handleClick: function() {
+      var elem = this.activeSection.getFocusedElement();
+      if (!elem) {
+        return;
+      }
+
+      switch(elem.id) {
+        case 'qrcode':
+          this._switchSection('big-qrcode');
+          break;
+        case 'config-icon':
+          this.config.prepareOptions((checked) => {
+            this._switchSection('config-section');
+            this.activeSection.focus(document.getElementById(checked));
+          });
+          break;
+        case 'option-pair-off':
+        case 'option-pair-on':
+        case 'option-disabled':
+          this.config.check(elem.id);
+          break;
+        case 'clear-paired-devices':
+          navigator.mozL10n.formatValue('clear-paired-devices-message')
+            .then((value) => {
+              if (confirm(value)) {
+                this.config.clearAuthorizedDevices();
+              }
+            });
+          break;
+        case 'save-config':
+          this.config.save();
+          // wait for settings change
+          setTimeout(this._switchSection.bind(this), 200);
+          break;
+      }
+    },
+
+    _handleBack: function() {
```

Comment:
We are simplifying page transition flow and assuming at most two layers of pages. Please add comments here as a warning.